osu.Game.Modes.Taiko/UI/TaikoHitRenderer.cs

```diff
@@ -3,17 +3,140 @@
 
 using osu.Game.Modes.Objects;
 using osu.Game.Modes.Objects.Drawables;
+using osu.Game.Modes.Taiko.Objects.Drawables;
 using osu.Game.Modes.Taiko.Objects;
 using osu.Game.Modes.UI;
+using osu.Framework.Graphics;
+using OpenTK;
+using osu.Game.Beatmaps;
+using osu.Framework.Allocation;
+using osu.Game.Beatmaps.Timing;
+using System.Collections.Generic;
 
 namespace osu.Game.Modes.Taiko.UI
 {
-    public class TaikoHitRenderer : HitRenderer<TaikoBaseHit>
+    public class TaikoHitRenderer : HitRenderer<TaikoHitObject>
     {
-        protected override HitObjectConverter<TaikoBaseHit> Converter => new TaikoConverter();
+        protected override HitObjectConverter<TaikoHitObject> Converter => new TaikoConverter();
 
-        protected override Playfield CreatePlayfield() => new TaikoPlayfield();
+        private Beatmap beatmap;
 
-        protected override DrawableHitObject GetVisualRepresentation(TaikoBaseHit h) => null;// new DrawableTaikoHit(h);
+        public TaikoHitRenderer(Beatmap beatmap)
+            : base(beatmap)
+        {
+            this.beatmap = beatmap;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            computeBarLines();
+        }
+
+        protected override Playfield CreatePlayfield() => new TaikoPlayfield()
+        {
+            RelativePositionAxes = Axes.Y,
+            Position = new Vector2(0, 0.4f)
+        };
+
+        protected override DrawableHitObject GetVisualRepresentation(TaikoHitObject h)
+        {
+            if (h is HitCircle)
+            {
+                switch (h.Type)

```

Comment:
h.Type should capture all of these. Avoid `is` whereever possible since casting (or checking for validity of a cast) is way more expensive than virtual method calls. In this case it's also less elegant, because we already have the virtual `h.Type`.